src/DynamoCore/Graph/Workspaces/WorkspaceModel.cs

```diff
@@ -436,6 +438,25 @@ protected virtual void OnSaving(XmlDocument obj)
             if (handler != null) handler(obj);
         }
 
+
+        /// <summary>
+        /// Event that is fired when the workspace is collecting used assemblies
+        /// </summary>
+        internal delegate HashSet<AssemblyName> CollectingAssembliesUsedHandler();
+        internal event CollectingAssembliesUsedHandler CollectingAssembliesUsed;
+
+        /// <summary>
+        /// Event that is fired when the workspace is collecting custom node package dependencies
+        /// </summary>
+        [Obsolete("This event will ignore subscribers other than PackageManagerExtension.")]
```

Comment:
I'm thinking the right way to handle this is to actually to use `InternalsVisibleToAttribute`